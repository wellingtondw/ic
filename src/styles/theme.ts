export default {
  border: {
    radius: '0.625rem'
  },
  font: {
    main: {
      family: "'Montserrat', sans-serif",
      normal: 400,
      bold: 700
    },
    secondary: {
      family: "'Lato', sans-serif",
      bold: 700
    },
    sizes: {
      xsmall: '0.75rem',
      small: '0.875rem',
      normal: '1rem',
      medium: '1.125rem',
      large: '2.25rem',
      xlarge: '4.5rem'
    }
  },
  colors: {
    primary: '#2C97D1',
    black: '#2A2A2A',
    blackBg: '#2B2B2B',
    white: '#FFFFFF',
    yellow: '#FBFE63'
  },
  spacings: {
    xxxsmall: '0.25rem', // 4px
    xxsmall: '0.5rem', //8px
    xsmall: '1rem', //16px
    small: '1.5rem', //24px
    medium: '2rem', // 32px
    large: '2.5rem', //40px
    xlarge: '3rem', //48px
    xxlarge: '3.5rem' //56px
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.1s ease-in-out'
  }
} as const
