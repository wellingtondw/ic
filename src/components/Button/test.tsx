import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'
import theme from '../../styles/theme'

import { Button } from '.'

describe('<Button />', () => {
  it('should be able to render correctly', () => {
    const { container } = renderWithTheme(
      <Button>choose your path again, Padawan</Button>
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should be able to render with primary variant', () => {
    renderWithTheme(<Button>choose your path again, Padawan</Button>)

    const buttonEl = screen.getByRole('button', {
      name: /choose your path again, Padawan/i
    })

    expect(buttonEl).toHaveStyle({
      background: theme.colors.primary,
      color: theme.colors.white,
      visibility: 'visible'
    })
  })

  it('should be able to render with dark variant', () => {
    renderWithTheme(
      <Button variant="dark">choose your path again, Padawan</Button>
    )

    const buttonEl = screen.getByRole('button', {
      name: /choose your path again, Padawan/i
    })

    expect(buttonEl).toHaveStyle({
      background: theme.colors.black,
      color: theme.colors.yellow,
      visibility: 'visible'
    })
  })

  it('should be able to render with light variant', () => {
    renderWithTheme(
      <Button variant="light">choose your path again, Padawan</Button>
    )

    const buttonEl = screen.getByRole('button', {
      name: /choose your path again, Padawan/i
    })

    expect(buttonEl).toHaveStyle({
      background: theme.colors.white,
      color: theme.colors.black,
      visibility: 'visible'
    })
  })

  it('should be able to render a disabled Button', () => {
    renderWithTheme(<Button disabled>choose your path again, Padawan</Button>)

    const buttonEl = screen.getByRole('button', {
      name: /choose your path again, Padawan/i
    })

    expect(buttonEl).toHaveStyle({
      opacity: 0.5,
      cursor: 'not-allowed',
      visibility: 'visible'
    })
  })

  it('should be able to render Button with loading', () => {
    renderWithTheme(<Button loading>choose your path again, Padawan</Button>)

    const contentEl = screen.getByText(/choose your path again, Padawan/i)
    const spinnerEl = screen.getByLabelText(/loading/i)

    expect(spinnerEl).toBeInTheDocument()
    expect(spinnerEl).toHaveStyle({
      position: 'absolute'
    })
    expect(contentEl).toHaveStyle({
      visibility: 'hidden'
    })
  })
})
