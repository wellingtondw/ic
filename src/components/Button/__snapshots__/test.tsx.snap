// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem 2rem;
  font-weight: bold;
  border: 0;
  border-radius: 0.625rem;
  position: relative;
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  background: #2C97D1;
  color: #FFFFFF;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0:hover:not(:disabled) {
  background: #298ec4;
}

.c1 {
  visibility: visible;
}

<button
  class="c0"
>
  <div
    class="c1"
  >
    choose your path again, Padawan
  </div>
</button>
`;
